src/nnet3/nnet-utils.cc

```diff
@@ -264,6 +280,57 @@ void SetLearningRate(BaseFloat learning_rate,
   }
 }
 
+void SetLearningRates(const Vector<BaseFloat> &learning_rates,
+                     Nnet *nnet) {
+  for (int32 c = 0, i = 0; c < nnet->NumComponents(); c++) {
+    Component *comp = nnet->GetComponent(c);
+    if (comp->Properties() & kUpdatableComponent) {
+      // For now all updatable components inherit from class UpdatableComponent.
+      // If that changes in future, we will change this code.
+      UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(comp);
+      if (uc == NULL)
+        KALDI_ERR << "Updatable component does not inherit from class "
+            "UpdatableComponent; change this code.";
+      KALDI_ASSERT(i < learning_rates.Dim());
+      uc->SetLearningRate(learning_rates(i));
+    }
+  }
+}
+
+void GetLearningRates(const Nnet &nnet, 
+                      Vector<BaseFloat> *learning_rates) {
+  learning_rates->Resize(NumUpdatableComponents(nnet));
+  for (int32 c = 0, i = 0; c < nnet.NumComponents(); c++) {
+    const Component *comp = nnet.GetComponent(c);
+    if (comp->Properties() & kUpdatableComponent) {
+      // For now all updatable components inherit from class UpdatableComponent.
+      // If that changes in future, we will change this code.
+      const UpdatableComponent *uc = dynamic_cast<const UpdatableComponent*>(comp);
+      if (uc == NULL)
+        KALDI_ERR << "Updatable component does not inherit from class "
+            "UpdatableComponent; change this code.";
+      (*learning_rates)(i++) = uc->LearningRate();
+    }
+  }
+}
+
+void ScaleNnetComponents(const Vector<BaseFloat> &scale_factors,
+                         Nnet *nnet) {
```

Comment:
OK, I just want to make sure all the code makes sense w.r.t. the current
interfaces.
I'm working on a commit to make the learning-rate interfaces easier.
Dan


On Sun, Mar 6, 2016 at 6:46 PM, Vimal Manohar <notifications@github.com>
wrote:

> In src/nnet3/nnet-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/560#discussion_r55151485>:
>
> > +  for (int32 c = 0, i = 0; c < nnet.NumComponents(); c++) {
> > +    const Component *comp = nnet.GetComponent(c);
> > +    if (comp->Properties() & kUpdatableComponent) {
> > +      // For now all updatable components inherit from class UpdatableComponent.
> > +      // If that changes in future, we will change this code.
> > +      const UpdatableComponent *uc = dynamic_cast<const UpdatableComponent*>(comp);
> > +      if (uc == NULL)
> > +        KALDI_ERR << "Updatable component does not inherit from class "
> > +            "UpdatableComponent; change this code.";
> > +      (*learning_rates)(i++) = uc->LearningRate();
> > +    }
> > +  }
> > +}
> > +
> > +void ScaleNnetComponents(const Vector<BaseFloat> &scale_factors,
> > +                         Nnet *nnet) {
>
> At the moment this program probably cannot be used with jesus layer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/560/files#r55151485>.
>